models/UserManager.php

```diff
@@ -0,0 +1,254 @@
+            FROM articles
+            JOIN admins
+            ON articles.writer=admins.email
+            WHERE posted='1'
+            ORDER BY date DESC
+        ");
+        
+        while($row = $req->fetch()){
+            
+            $admin   = new User();
+            $admin   ->setName($row['name']);
+                        
+            $admins[] = $admin;
+        }
+        return $admins;
+    }
+    
+    // Fonction permettant de vérifier que la session est bien active sous le role administrateur
+    function admin(){
+        if(isset($_SESSION['admin'])){
+            $tableau = [
+                'email'     =>  $_SESSION['admin'],
+                'role'      =>  'admin'
+            ];
+
+            $sql = "SELECT * FROM admins WHERE email=:email AND role=:role";
+            $req = $this->db->prepare($sql);
+            $req->execute($tableau);
+            $exist = $req->rowCount($sql);
+
+            return $exist;
+            
+        }else{
+            $exist = 0;
+            return $exist;
+        }
+    }
+
+    // Fonction qui vérifie qu'un utilisateur est bien administrateur
+    function is_admin($email,$password){
+        $tableau = [
+            'email'     =>  $email,
+            'password'  =>  sha1($password)
+        ];
+        $sql = "SELECT * FROM admins WHERE email = :email AND password = :password";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        $exist = $req->rowCount($sql);
+        
+        return $exist;
+    }
+
+    // Fonction permettant de savoir si un administrateur a un mot de passe
+    function hasnt_password(){
+        $sql = "SELECT * FROM admins WHERE email = '{$_SESSION['admin']}' AND password = ''";
+        $req = $this->db->prepare($sql);
+        $req->execute();
+        $exist = $req->rowCount($sql);
+        
+        return $exist;
+    }
+
+    // Fonction qui gère les settings
+    function email_taken($email){
+        $tableau = ['email'   =>  $email];
+        $sql = "SELECT * FROM admins WHERE email = :email";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        $free = $req->rowCount($sql);
+
+        return $free;
+    }
+
+    //Fonction permettant de créer un token
+    function token($length){
+        $chars = "azertyuiopqsdfghjklmwxcvbnAZERTYUIOPQSDFGHJKLMWXCVBN0123456789";
+        
+        return substr(str_shuffle(str_repeat($chars,$length)),0,$length);
+    }
+
+    // Fonction permettant d'ajouter un modérateur / administrateur et de lui envoyer un mail
+    function add_modo($name,$email,$role,$token){
+        $tableau= [
+            'name'      =>  $name,
+            'email'     =>  $email,
+            'token'     =>  $token,
+            'role'      =>  $role
+        ];
+
+        $sql = "INSERT INTO admins(name,email,token,role) VALUES(:name,:email,:token,:role)";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Detected usage of a non-sanitized, non-validated input variable $_GET:             WHERE articles.id = '{$_GET['id']}'
](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)